e2e/node/acp-v3.test.ts

```diff
@@ -0,0 +1,271 @@
+      read: true,
+      append: false,
+      write: false,
+    });
+
+    let selfRule = acp.createRule(policyResourceUrl + "#rule-self");
+    selfRule = acp.addAgent(selfRule, session.info.webId!);
+    // This policy denies write access to the current user,
+    // but allows write access so the Resource can still be removed afterwards:
+    let selfWriteNoReadPolicy = acp.createPolicy(
+      policyResourceUrl + "#policy-selfWriteNoRead"
+    );
+    selfWriteNoReadPolicy = acp.addAllOfRuleUrl(
+      selfWriteNoReadPolicy,
+      selfRule
+    );
+    selfWriteNoReadPolicy = acp.setAllowModes(selfWriteNoReadPolicy, {
+      read: false,
+      append: true,
+      write: true,
+    });
+    selfWriteNoReadPolicy = acp.setDenyModes(selfWriteNoReadPolicy, {
+      read: true,
+      append: false,
+      write: false,
+    });
+
+    let policyResource = createSolidDataset();
+    policyResource = setThing(policyResource, publicRule);
+    policyResource = setThing(policyResource, publicReadPolicy);
+    policyResource = setThing(policyResource, selfRule);
+    policyResource = setThing(policyResource, selfWriteNoReadPolicy);
+
+    return saveSolidDatasetAt(policyResourceUrl, policyResource, options);
+  }
+
+  async function applyPolicyToPolicyResource(
+    resourceUrl: UrlString,
+    policyUrl: UrlString,
+    session: Session
+  ) {
+    const resourceWithAcr = await acp.getSolidDatasetWithAcr(resourceUrl, {
+      fetch: session.fetch,
+    });
+    if (!acp.hasAccessibleAcr(resourceWithAcr)) {
+      throw new Error(
+        `The test Resource at [${getSourceUrl(
+          resourceWithAcr
+        )}] does not appear to have a readable Access Control Resource. Please check the Pod setup.`
+      );
+    }
+    const changedResourceWithAcr = acp.addPolicyUrl(
+      resourceWithAcr,
+      policyUrl
+    );
+    return acp.saveAcrFor(changedResourceWithAcr, options);
+  }
+
+  it("can deny Read access", async () => {
+    // Create a Resource containing Access Policies and Rules:
+    await initialisePolicyResource(sessionResource, session);
+
+    // Verify that we can fetch the Resource before Denying Read access:
+    await expect(
+      getSolidDataset(sessionResource, options)
+    ).resolves.not.toBeNull();
+
+    // In the Resource's Access Control Resource, apply the Policy
+    // that just so happens to be defined in the Resource itself,
+    // and that denies Read access to the current user:
+    await applyPolicyToPolicyResource(
+      sessionResource,
+      sessionResource + "#policy-selfWriteNoRead",
+      session
+    );
+
+    // Verify that indeed, the current user can no longer read it:
+    await expect(
+      getSolidDataset(sessionResource, options)
+    ).rejects.toThrow(
+      // Forbidden:
+      // @ts-ignore-next
+      expect.objectContaining({ statusCode: 403 }) as FetchError
+    );
+
+    // Clean up:
+    await deleteSolidDataset(sessionResource, options);
```

Comment:
I think this is redundant with the `afterEach` block.

Reply:
Removed all redundant cleanup in https://github.com/inrupt/solid-client-js/pull/1426/commits/39b56eba3a4e1260fe98ba5d99d87eba83d4c99a.